src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerPanel.tsx

```diff
@@ -75,11 +89,45 @@ export const TemplatesCatalogDrawerPanel: React.FC<TemplatesCatalogDrawerPanelPr
     const interfaces = getTemplateInterfaces(template);
     const disks = getTemplateDisks(template);
     const isDefaultTemplate = isDefaultVariantTemplate(template);
-    const { memory, cpuCount } = getTemplateFlavorData(template);
     const hostDevicesCount = getHostDevices(vmObject)?.length || 0;
     const gpusCount = getGPUDevices(vmObject)?.length || 0;
     const hardwareDevicesCount = hostDevicesCount + gpusCount;
 
+    const [updatedVMObject, setUpdatedVMObject] = useState<V1VirtualMachine>(undefined);
+
+    useEffect(() => {
+      k8sCreate<V1Template>({
+        model: ProcessedTemplatesModel,
+        data: setValueForRequiredParams(template),
+        queryParams: {
+          dryRun: 'All',
+        },
+      }).then((processedTemplate) => {
+        setUpdatedVMObject(getTemplateVirtualMachineObject(processedTemplate));
+      });
```

Comment:
any specific reason to run this on init render and not on submit function?
you are in a race condition here if the template wasn't loaded on the initial render you will still have undefined in your updatedVMObject

Reply:
Just a note, I needed to process the template to be able to use the VM object that is suitable for using it with our context hook for saving the VM data and being able to get the data later if user decides to customize the VM. And also to be able to use the CPU/Mem modal we already have implemented. I expect that more stuff is gonna be soon editable in the drawer, so putting everything to url won't be a good idea anymore (and IMHO it never was a good idea but we do it for some data already anyway).

Reply:
> What I can see here I am just missing adding the case when the template doesn't load or process successfully on the first render. WDYT?

```
useEffect(() => {
  if(loaded && isEmpty(loadError) && !isEmpty(template)) {
    k8sCreate<V1Template>({
      model: ProcessedTemplatesModel,
      data: setValueForRequiredParams(template),
      queryParams: {
        dryRun: 'All',
       },
      }).then((processedTemplate) => {
        setUpdatedVMObject(getTemplateVirtualMachineObject(processedTemplate));
      });
}, [loaded, loadError, template]);
```
adding a condition should do the trick - the useEffect will run a few times but will process the template only once because of the condition which guarantees that the template is loaded without an error

Reply:
Hey,

1. we better not use "// eslint-disable-next-line react-hooks/exhaustive-deps"
as next react versions will be without a dependency array and will auto-add dependency, and it will cause bugs. better just add more protection to the code block and all needed dependencies. 

2. this file is getting very long already, about 250 lines of code. Those changes also add 50+ lines of code just for this file, please export functions and inner components to different utils/components files.

3. u are importing VirtualMachineDescriptionItem which belongs to VM pages into catalog, if you would like to use it please move VirtualMachineDescriptionItem into a more proper location under our shared folder. The same goes to CPUMemory component.

4. why do u need to process the template before any change has done at all?
you are getting the template as a prop which is good, the proccessedtemplate should just check the values that have been changed are ok. why processing without any value change?

Reply:
1,2,3 yes.

4. because without that we wont be able to use the existing cpu/mem modal and also the context hook. the option to avoid that is to put the data to the url, which I am not going to do.

Reply:
The thing with URL is that it refresh resistance, and will keep the user flow with the same data when the page is refreshing. If I'm not mistaken, the context is also saved in localstroage so we have some good behavior that we want, just make sure your changes are refresh resistance. 

processedTemplate- gives u a VM with params set in VM? That's what u are looking for?

Reply:
Seems that processing a template does more than just setting some params or generating a VM name (which is not interesting for us now). It makes the VM object a "proper" VM object, usable for the context hook and for `CPUMemoryModal` component. I simply cannot use `updateVM` with the VM object taken from a template if the template wasn't already processed. It knows that it isn't really a VM.

Reply:
Before this addition, we also used ProcessedTemplates for editing, and updaing the name of VM which is also available in the drawer? should work the same way ?

Reply:
My solution for updating the name in the drawer did require processing the template for sure. However, somebody has changed that to putting the name to url to avoid that (and there's more data in the url, when being in the drawer.. data that could be saved and used later by the same context hook I am using in this PR. We could use the hook for all the data from drawer, not just this. Just the code around is not super short, but we could figure that out.. ).